compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1466,26 +1457,30 @@ public boolean isInherentlyImmutableType(BType type) {
             case TypeTags.HANDLE:
                 return true;
             case TypeTags.XML:
-                return ((BXMLType) type).constraint.tag == TypeTags.NEVER;
-            case TypeTags.TYPEREFDESC:
-                return isInherentlyImmutableType(((BTypeReferenceType) type).referredType);
+                return getReferredType(((BXMLType) type).constraint).tag == TypeTags.NEVER;
         }
         return false;
     }
 
     public static BType getReferredType(BType type) {
```

Comment:
Is this name still accurate? It's no longer just referred type?

Reply:
any suggestions?

Reply:
Replaced with `getConclusiveType`